### PR TITLE
D3ASIM-3442: Added tolerance to floating point comparison for the off…

### PR DESCRIPTION
**Pull request by spyrostz**
…er energy and the to-be-accepted energy.

### CODE REVIEW

**Review by hannesdiedrich — APPROVED** [review]
Did you validate that it is a floating point issue or is this PR just for checking out whether this is the issue ?

**Comment by spyrostz** [comment]
> Did you validate that it is a floating point issue or is this PR just for checking out whether this is the issue ?

Of course, I tested it by connecting to a clone of the dev DB, which had another configuration that exhibited the same problem (configuration name is GZ, and you can see the same logs as the bug reports in Kibana - reason why I wanted the dev DB was because the DB there is much smaller in size and quicker to clone compared to production). Reproduced the issue first, the issue indeed appeared, then checked out the energy values, validated that they were within the range of the floating point precision, applied the fix, rerun the same configuration again and no errors appeared anymore.   
TLDR, this fix should solve the issue in production as is.

**Review by faizan2590 — APPROVED** [review]
